lib/index.js

```diff
@@ -139,11 +154,25 @@ class Linter {
     return false;
   }
 
+  getRulesForOverrides(overrides) {
+    if (overrides.length) {
+      return overrides.map(item => {
+        const fileArr = Array.from(expandFileGlobs(item.files, true));
+        item.files = fileArr.length ? fileArr : item.files;
+        return item;
+      });
+    }
+    return [];
+  }
+
   verify(options) {
     let messages = [];
     let pendingStatus = this.statusForModule('pending', options.moduleId);
     let shouldIgnore = this.statusForModule('ignore', options.moduleId);
 
+    let overrides = JSON.parse(JSON.stringify(this.config.overrides));
+    let overrideRules = this.getRulesForOverrides(overrides);
```

Comment:
Hmm, I don't think we want to glob the file system for every override. In fact, I don't think we actually should hit `globby` at all in this case.  For very large projects that have a decent number of `overrides`, this would currently result in a non-trivial slow down (due to many many more I/O operations).

Specifically, every time `verify` is called we know the full path that we are linting and there is no reason to run globby again to see if a given full path is included within a given array of globs. Take a look at [micromatch's isMatch](https://github.com/micromatch/micromatch#ismatch) function, which should let us do what we want without additional I/O.